backend/src/users/user-relationship/entities/user-relationship.entity.ts

```diff
@@ -15,25 +15,25 @@ export enum validRelationships {
 }
 
 @Entity('user_relationships')
-@Check(`"source_id" <> "target_id"`) // 				makes [2,2] impossible
+@Check(`"source" <> "target"`) // 				makes [2,2] impossible
 export class UserRelationship {
 	@PrimaryGeneratedColumn()
 	id: number;
 
 	@ManyToOne(() => User, (user: User) => user.relationshipSource, {
 		eager: true,
 	})
-	@JoinColumn({ name: 'source_id' })
-	source_id: User;
+	@JoinColumn({ name: 'source' })
+	source: User;
 
 	@ManyToOne(() => User, (user: User) => user.relationshipTarget, {
 		eager: true,
 	})
-	@JoinColumn({ name: 'target_id' })
-	target_id: User;
+	@JoinColumn({ name: 'target' })
+	target: User;
 
-	// @Column()
-	// specifier_id: number;
+	@Column()
+	specifier_id: number;
```

Comment:
soooooooo I figured out that it is used and how

If A blocks B
a relationship between A and B is made (A=source, B=target)
if A unblocks B, the relationship is updated to type `none`
if B blocks A, the same relationship is updated to `blocked` again (and specifier_id is set) The specifier ID is then used to determine who blocked who, and not the source, since it's the same relationship reused.

i'm now thinking that it could have been done by removing the old relationship when you unblock. But this works, so alll goood :))

Reply:
exactly, as soon as a rel exists, it is only ever updated. indeed via delete and then create is just as viable